src/ecc.rs

```diff
@@ -62,6 +144,66 @@ impl CommonKeyFactory for ECCPubFactory {}
 
 impl PubKeyFactory for ECCPubFactory {}
 
+type Version = u64;
+
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+enum ECParameters<'a> {
+    // ecParametdders   ECParameters,
+    OId(asn1::ObjectIdentifier),
+    // implicitlyCA asn1::Null,
+    CurveName(asn1::PrintableString<'a>),
+}
+
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+struct ECPrivateKey<'a> {
+    version: Version,
+    private_key: &'a [u8],
+    #[explicit(0)]
+    parameters: Option<ECParameters<'a>>,
+    #[explicit(1)]
+    public_key: Option<DerEncBigUint<'a>>,
+}
+
+impl ECPrivateKey<'_> {
+    pub fn new_owned<'a>(
+        private_key: &'a Vec<u8>,
+    ) -> KResult<ECPrivateKey<'a>> {
+        Ok(ECPrivateKey {
+            version: 1,
+            private_key: private_key.as_slice(),
+            parameters: None,
+            public_key: None,
+        })
+    }
+}
+
+#[derive(asn1::Asn1Read, asn1::Asn1Write)]
+struct PrivateKeyInfo<'a> {
+    version: Version,
+    private_key_algorithm: asn1::ObjectIdentifier,
+    // private_key: &'a [u8],
+    private_key: asn1::OctetStringEncoded<ECPrivateKey<'a>>,
+    // likely not needed
+    //#[explicit(1)]
```

Comment:
needed if we ever want to parse data coming from outside our module.

Reply:
we can simply ignore the content of course, but the asn.1 structure needs to be complete or the parser will just fail

Reply:
Added them, but hope we will not have to use them any time soon :)